typescript/infra/scripts/deploy.ts

```diff
@@ -241,6 +250,16 @@ async function main() {
     config = {
       ethereum: coreConfig.ethereum.defaultHook,
     };
+  } else if (module === Modules.CCIP) {
+    if (environment !== 'mainnet3') {
+      throw new Error('CCIP is only supported on mainnet3');
+    }
+    config = getCCIPDeployConfig(filteredTargetNetworks);
+    deployer = new HyperlaneCCIPDeployer(
+      multiProvider,
+      getEnvAddresses(environment),
+      contractVerifier,
+    );
```

Comment:
with the new design, the intention is that these hooks/isms are reused for each (origin,destination) pair. As we don't really want users to deploy these themselves (they'll just use the lookups), I kept the config generation + intended deployer usage in infra